libethereum/State.h

```diff
@@ -247,11 +250,18 @@ class State
 	void noteAccountStartNonce(u256 const& _actual);
 
 private:
-	/// Retrieve all information about a given address into the cache.
-	/// If _requireMemory is true, grab the full memory should it be a contract item.
-	/// If _forceCreate is true, then insert a default item into the cache, in the case it doesn't
-	/// exist in the DB.
-	void ensureCached(Address const& _a, bool _requireCode, bool _forceCreate) const;
+	/// @returns the account at the given address or a null pointer if it does not exist.
+	/// The pointer is valid until the next access to the state or account.
+	Account const* account(Address const& _a, bool _requireCode = false) const;
+
+	/// @returns the account at the given address or a null pointer if it does not exist.
+	/// The pointer is valid until the next access to the state or account.
+	Account* account(Address const& _a, bool _requireCode = false);
+
+	/// Retrieve certain information about a given address into the cache and return
+	/// a pointer to the relevant account (or a null pointer if it does not exist).
+	/// If _requireCode is true, also load the code.
+	Account* ensureCached(Address const& _a, bool _requireCode) const;
```

Comment:
I would prefer a solution with only two functions, but I think it is important to distinguish between const and non-const access to the accounts. This might save us another bug.